File: core/src/test/java/w/commander/execution/TestCommandExecutionContextFactory.java
package w.commander.execution;

import w.commander.CommandSender;
import w.commander.RawCommandArguments;

/**
 * @author whilein
 */
public class TestCommandExecutionContextFactory implements CommandExecutionContextFactory {
    @Override
    public CommandExecutionContext create(CommandSender sender, CommandExecutor executor, RawCommandArguments arguments) {
        return new TestCommandExecutionContext(sender, executor, arguments);
    }
}

File: tests/src/main/kotlin/w/commander/tests/TestCommandExecutionContextFactory.kt
package w.commander.tests

import w.commander.CommandSender
import w.commander.RawCommandArguments
import w.commander.execution.CommandExecutionContext
import w.commander.execution.CommandExecutionContextFactory
import w.commander.execution.CommandExecutor

/**
 * @author whilein
 */
class TestCommandExecutionContextFactory : CommandExecutionContextFactory {
    override fun create(
            sender: CommandSender,
            executor: CommandExecutor,
            arguments: RawCommandArguments
    ): CommandExecutionContext {
        return TestCommandExecutionContext(sender, executor, arguments)
    }
}

